src/relentless/simulate/lammps.py

```diff
@@ -555,30 +607,28 @@ def __init__(self, steps, timestep, T, friction, seed, analyzers):
         self.seed = seed
 
     def to_commands(self, sim):
-        # obtain per-type mass (arrays 1-indexed using lammps convention)
-        Ntypes = len(sim.types)
-        mass = sim[sim.initializer]["_lammps"].numpy.extract_atom("mass")
-        if mass is None or mass.shape != (Ntypes + 1,):
-            raise ValueError("Per-type masses not set.")
-        mass = numpy.squeeze(mass)
-
         # obtain per-type friction factor
-        friction = numpy.zeros_like(mass)
+        Ntypes = len(sim.types)
+        mass = numpy.zeros(Ntypes)
+        friction = numpy.zeros(Ntypes)
         for t in sim.types:
+            typeidx = sim[sim.initializer]["lammps_types"][t] - 1
             try:
-                friction[sim[sim.initializer]["lammps_types"][t]] = self.friction[t]
+                friction[typeidx] = self.friction[t]
             except TypeError:
-                friction[sim[sim.initializer]["lammps_types"][t]] = self.friction
+                friction[typeidx] = self.friction
             except KeyError:
                 raise KeyError("The friction factor for type {} is not set.".format(t))
 
+            mass[typeidx] = sim.masses[t]
+
         # compute per-type damping parameter and rescale if multiple types
         damp = numpy.divide(mass, friction, where=(friction > 0))
-        damp_ref = damp[1]
+        damp_ref = damp[0]
         if Ntypes > 1:
             scale = damp / damp_ref
             scale_str = " ".join(
-                ["scale {} {}".format(i + 1, s) for i, s in enumerate(scale[2:])]
+                ["scale {} {}".format(i + 1, s) for i, s in enumerate(scale[1:])]
```

Comment:
This might need to be `i+2` to account for the shift by 1 for the index, plus another 1 for the slice.
```suggestion
                ["scale {} {}".format(i + 2, s) for i, s in enumerate(scale[1:])]
```